plugins/rapture/nexus-coreui-plugin/src/main/resources/static/rapture/NX/coreui/controller/Repositories.js

```diff
@@ -371,6 +375,45 @@ Ext.define('NX.coreui.controller.Repositories', {
           scope: button
         }
     );
+  },
+
+  /**
+   * @override
+   * Rebuild Repository index.
+   */
+  rebuildIndex: function(model) {
+    var me = this;
+
+    NX.direct.coreui_Repository.rebuildIndex(model.getId(), function(response) {
+      me.getStore('Repository').load();
+      if (Ext.isObject(response) && response.success) {
+        NX.Messages.add({ text: 'Repository index rebuilt: ' + me.getDescription(model), type: 'success' });
+      }
+    });
+  },
+
+  /**
+   * @private
+   * Enable 'Rebuild Index' when user has 'read' permission and task is 'runnable'.
+   */
+  bindRebuildIndexButton: function(button) {
+    var permittedCondition;
+    button.mon(
+        NX.Conditions.and(
+            permittedCondition = NX.Conditions.isPermitted('nexus:repository-admin:*:*:delete'),
```

Comment:
not sure why we 'and' these conditions, since either/or looks like it should enable the action: generic for any repo OR specific to this particular repo